src/pool/srv_target.c

```diff
@@ -107,6 +107,84 @@ ds_pool_child_purge(struct pool_tls *tls)
 	}
 }
 
+static void
+gc_ult(void *arg)
+{
+	struct ds_pool_child	*child = (struct ds_pool_child *)arg;
+	struct dss_module_info	*dmi = dss_get_module_info();
+	int			 rc;
+
+	D_DEBUG(DF_DSMS, DF_UUID"[%d]: GC ULT started\n",
+		DP_UUID(child->spc_uuid), dmi->dmi_tgt_id);
+
+	D_ASSERT(child->spc_gc_req != NULL);
+	while (!dss_ult_exiting(child->spc_gc_req)) {
+
+		rc = vos_gc_pool_run(child->spc_hdl, -1, dss_ult_yield,
+				     (void *)child->spc_gc_req);
+		if (rc < 0)
+			D_ERROR(DF_UUID"[%d]: GC pool run failed. "DF_RC"\n",
+				DP_UUID(child->spc_uuid), dmi->dmi_tgt_id,
+				DP_RC(rc));
+
+		if (dss_ult_exiting(child->spc_gc_req))
+			break;
+
+		/* Sleep 2 seconds */
+		sched_req_sleep(child->spc_gc_req, 2ULL * 1000);
```

Comment:
I think this should be improved in the future, e.g: based on SCM space pressure, I/O workload...

Reply:
Yes, I think ideally the GC and aggregation ULT (especially GC ULT) should always in sleep unless it's woke up by somebody else (on container destroy or aggregation progressed). I'm going to enlarge this sleep period in follow-on patch.